lib/ansible/constants.py

```diff
@@ -119,7 +119,8 @@ def __getitem__(self, y):
 VAULT_VERSION_MAX = 1.0
 
 # This matches a string that cannot be used as a valid python variable name i.e 'not-valid', 'not!valid@either' '1_nor_This'
-INVALID_VARIABLE_NAMES = re.compile(r'^[^a-zA-Z_]|[^a-zA-Z0-9_]')
+INVALID_VARIABLE_NAMES = re.compile(r'^[\d\W]|[^\w]')
```

Comment:
Looks good. This fixes identifying valid python-3 unicode variables as invalid characters for me. cc @kdelee since I think you were testing this.

Reply:
it was @felixfontein that brought it up iirc, in irc

Reply:
Oh. @kdelee brought it up in slack too

Reply:
resmo then? i created test from irc conversation this morning, that is what i've been using as lithmus